compiler/src/iree/compiler/Dialect/Stream/Transforms/ScheduleAllocation.cpp

```diff
@@ -56,12 +56,19 @@ static void computeRegionValueAliases(Operation *regionOp,
     aliasedSet.insert(streamValue);
   };
 
+  // Filter out to only resource results - some regions may return additional
+  // things like stream.async.execute returning a timepoint.
+  auto resourceResults = llvm::to_vector_of<OpResult>(
+      llvm::make_filter_range(regionOp->getResults(), [](OpResult result) {
+        return result.getType().isa<IREE::Stream::ResourceType>();
+      }));
+
   // Start with outputs so that we handle tied values that may lead all the way
   // back up the chain to the stream inputs.
   auto tiedStreamOp = cast<IREE::Util::TiedOpInterface>(regionOp);
   auto yieldOp = cast<IREE::Stream::YieldOp>(block->getTerminator());
   for (auto it :
-       llvm::zip(regionOp->getResults(), yieldOp.getResourceOperands())) {
+       llvm::zip_equal(resourceResults, yieldOp.getResourceOperands())) {
     auto outerResult = std::get<0>(it);
     auto innerResult = std::get<1>(it);
```

Comment:
We can use `structured binding declaration` for this kind of iteration. E.g.,

```
for (auto [outerResult, innerResult] : llvm::zip_equal(resourceResults, yieldOp.getResourceOperands())) {
```

note that they are captured by reference, IIRC.